project/new-evaluation.json

```diff
@@ -0,0 +1,247 @@
+{ "items":  [
+    {
+        "legend": "OOP, Design Patterns",
+        "dropDownOptions": [
+            [
+                {
+                    "label": "Classes",
+                    "options": ["0", "1", "2", "3"]
```

Comment:
Since these `option`s get repeated all over the place, might as well abstract them away into a centralized location for reuse, thus keeping your config object DRY as well.

Reply:
Hm.. how exactly could it be done ?

Reply:
What if we were to remove the options from item level and move them to the top level of the configuration object? Could we then update the rendering function to take in the item to render, along with these top-level options, and have it create the corresponding markup based on the two now disjunct pieces of information? I bet we could, if we wanted to! And I bet it would amount to a lighter config object, as a result! :P

Reply:
Ok, I have no choice :)) I'll try it 👍

Reply:
Done. I hope it's what you've thought of. See last commit.

Reply:
You could even move `options` a couple of levels higher (i.e., on the same level as `items`), but even with this change, the config object has become much more light-weight, don't you think? :)

As a future enhancement, if we wanted even more customization ability, we could provide an `options` configuration for each of these levels, and default to the previous level if none is specified for the current level. For example, if we have `options` for the specific item, use that. If it's missing, but we have `options` for the category, then use that. Otherwise, if that's missing as well, use whatever `options` we have available for the top-level configuration object. But that's only one of possible future improvements, as we don't really need that much customization at this point in time. :wink: